src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,42 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(victim) {
+    if (victim instanceof Herbivore && victim.hidden === false) {
+      victim.health -= 50;
+
+      if (victim.health <= 0) {
+        Animal.removeDeadAnimals();
```

Comment:
[CODE PERFORMANCE] - Instead of checking the victim's health and then calling the `removeDeadAnimals` method, consider filtering the `Animal.alive` array directly within the `bite` method after reducing the health. This will avoid the need for a conditional check and will make the code cleaner and potentially more performant.